### PR TITLE
Move mtls package from internal to pkg

### PR DESCRIPTION
**Pull request by eloycoto**
Two commits with the description. 

The main use case is to be able to use some functions on the device and do not duplicate things at all.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -303,7 +303,10 @@ func main() {
 					if r.TLS != nil {
 						authType := yggdrasilAPIHandler.GetAuthType(r)
 						if !mtls.VerifyRequest(r, authType, opts, CACertChain) {
-							w.WriteHeader(http.StatusUnauthorized)
+							// w.WriteHeader(http.StatusUnauthorized)
+							// return
```
Comment (gciavarrini) [inline main.go:307]: Are these comments necessary?

File: pkg/mtls/caprovider_secret.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ const (
 	clientCertCertKey   = "client.crt"
 
 	certOrganization       = "flotta-operator"
-	certRegisterCN         = "register"
+	CertRegisterCN         = "register"
```
Comment (gciavarrini) [inline pkg/mtls/caprovider_secret.go:35]: Why should this const be exported? Will it be used in the device?

File: pkg/mtls/cert_utils.go [hunk 3]
```diff
@@ -106,9 +106,9 @@ func (c *CertificateGroup) CreatePem() error {
 	return nil
 }
 
-func (c *CertificateGroup) marshalKeyToPem() (*bytes.Buffer, error) {
+func (c *CertificateGroup) MarshalKeyToPem(privKey crypto.PrivateKey) (*bytes.Buffer, error) {
 	privKeyPEM := new(bytes.Buffer)
-	switch t := c.privKey.(type) {
+	switch t := privKey.(type) {
```
Comment (gciavarrini) [inline pkg/mtls/cert_utils.go:111]: Question to learn: what is the reason of this change?

File: pkg/mtls/cert_utils.go [hunk 4]
```diff
@@ -163,6 +163,10 @@ func (c *CertificateGroup) GetCert() *x509.Certificate {
 	return c.cert
 }
 
+func (c *CertificateGroup) GetKey() crypto.PrivateKey {
```
Comment (gciavarrini) [inline pkg/mtls/cert_utils.go:166]: Why do not export the `privKey` fields instead?

File: pkg/mtls/caprovider_secret.go (hunk 2 above)
Comment (eloycoto) [reply pkg/mtls/caprovider_secret.go:35]: yes

File: main.go (hunk 1 above)
Comment (eloycoto) [reply main.go:307]: When ECOPROJECT task will be in there, I'll uncomment, but I do not want to break the CI too much on device-changes.

Comment (masayag) [reply main.go:307]: perhaps worth adding FIXME or TODO comment to make this obvious (and avoid unintentional clean-up PRs by others)

**Review by masayag — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (eloycoto) [reply main.go:307]: Is just in the next line

Comment (masayag) [reply main.go:307]: :-) How did I miss it...

**Review by pkliczewski — APPROVED** [review]
